spec/lib/whitehall_importer/create_image_revision_spec.rb

```diff
@@ -0,0 +1,110 @@
+# frozen_string_literal: true
+
+RSpec.describe WhitehallImporter::CreateImageRevision do
+  describe "#call" do
+    let(:img_url) { "https://assets.publishing.service.gov.uk/government/uploads/valid-image.jpg" }
+    let(:img_file) { "960x640.jpg" }
+    let(:whitehall_image) do
+      { "url": img_url, "caption": "Caption", "alt_text": "Alt text" }.with_indifferent_access
+    end
+
+    before do
+      stub_request(:get, img_url).to_return(
+        status: 200,
+        body: file_fixture(img_file).read,
+      )
+    end
+
+    context "A valid image is provided" do
+      before do
+        stub_request(:get, img_url).to_return(status: 200, body: file_fixture("960x640.jpg"))
+      end
+
+      it "should create an Image::MetadataRevision that preserves the alt text and caption" do
+        described_class.call(whitehall_image)
+        expect(Image::MetadataRevision.count).to eq(1)
+        expect(Image::MetadataRevision.first.caption).to eq(whitehall_image["caption"])
+        expect(Image::MetadataRevision.first.alt_text).to eq(whitehall_image["alt_text"])
```

Comment:
Minor and just my personal opinion. I personally avoid tests that expect the db to empty as these are with `.count` and `.first` as these end up being coupled to the db state which can make the tests fragile.

My order of preference with these is:

1) ideally test code either changes or returns something, then no risks at all about initial state or any concurrency.
2) test last item in the db, which is vulnerable to concurrency but doesn't require initial state. 

For these tests I think your most idiomatic situation would be testing the image revision that is returned rather than looking one up from the db.

Reply:
Have removed references to `.count` and `.first` 👍